lightning/src/chain/channelmonitor.rs

```diff
@@ -2744,19 +2744,32 @@ impl<Signer: EcdsaChannelSigner> ChannelMonitorImpl<Signer> {
 		// Treat the sweep as urgent as long as there is at least one HTLC which is pending on a
 		// valid commitment transaction.
 		if !self.current_holder_commitment_tx.htlc_outputs.is_empty() {
-			return ConfirmationTarget::UrgentOnChainSweep;
+			let minimum_expiry = self.current_holder_commitment_tx.htlc_outputs
+				.iter()
+				.map(|o| o.0.cltv_expiry)
+				.min();
+			return ConfirmationTarget::UrgentOnChainSweep(minimum_expiry);
 		}
 		if self.prev_holder_signed_commitment_tx.as_ref().map(|t| !t.htlc_outputs.is_empty()).unwrap_or(false) {
-			return ConfirmationTarget::UrgentOnChainSweep;
+			let minimum_expiry = self.prev_holder_signed_commitment_tx.as_ref().map(|t| t.htlc_outputs
+				.iter()
+				.map(|o| o.0.cltv_expiry)
+				.min()
+			).flatten();
+			return ConfirmationTarget::UrgentOnChainSweep(minimum_expiry);
```

Comment:
This current logic stops evaluating as soon as any HTLCs are found on a single commitment tx.

For force closing, we need to find the minimum expiry across *all* active commitment transactions, not just on the first one we find HTLCs on.